pkg/bootstrap/platform/gcp.go

```diff
@@ -207,19 +210,52 @@ func parseGCPMetadata() (pid, npid, cluster, location string) {
 	return envPid, envNpid, envCluster, envLocation
 }
 
+func isZone(z string) bool {
+	return zonePattern.MatchString(z)
+}
+
 // Converts a GCP zone into a region.
 func zoneToRegion(z string) (string, error) {
-	// Zones are in the form <region>-<zone_suffix>, so capture everything but the suffix.
-	re := regexp.MustCompile("(.*)-.*")
-	m := re.FindStringSubmatch(z)
-	if len(m) != 2 {
-		return "", fmt.Errorf("unable to extract region from GCP zone: %s", z)
+	if !isZone(z) {
+		return "", fmt.Errorf("invalid zone: %q", z)
 	}
-	return m[1], nil
+	zoneSuffixIdx := strings.LastIndex(z, "-")
+	return z[:zoneSuffixIdx], nil
 }
 
 // Locality returns the GCP-specific region and zone.
 func (e *gcpEnv) Locality() *core.Locality {
+	if e.locality != nil {
```

Comment:
nit: would once.Do be any simpler here? this is fine too but it seems like that is the intended use case of once.Do (although its not that user friendly :shrug: )

Reply:
Adopted your suggestion. The whole thing needs a larger overhaul, but I'm nervous about changing too much at the moment with zone/region identification.